packages/auto-approve/test/owlbot-template-changes.test.ts

```diff
@@ -0,0 +1,199 @@
+      3,
+      [{filename: 'hello', sha: '2345'}],
+      'testRepoName',
+      'testRepoOwner',
+      1,
+      octokit,
+      'body'
+    );
+
+    const fileRequest = nock('https://api.github.com')
+      .get('/repos/testRepoOwner/testRepoName/contents/.repo-metadata.json')
+      .reply(200, {
+        name: '.repo-metadata.json',
+        content:
+          'ewogICAgIm5hbWUiOiAiZGxwIiwKICAgICJuYW1lX3ByZXR0eSI6ICJDbG91ZCBEYXRhIExvc3MgUHJldmVudGlvbiIsCiAgICAicHJvZHVjdF9kb2N1bWVudGF0aW9uIjogImh0dHBzOi8vY2xvdWQuZ29vZ2xlLmNvbS9kbHAvZG9jcy8iLAogICAgImNsaWVudF9kb2N1bWVudGF0aW9uIjogImh0dHBzOi8vY2xvdWQuZ29vZ2xlLmNvbS9weXRob24vZG9jcy9yZWZlcmVuY2UvZGxwL2xhdGVzdCIsCiAgICAiaXNzdWVfdHJhY2tlciI6ICIiLAogICAgInJlbGVhc2VfbGV2ZWwiOiAiZ2EiLAogICAgImxhbmd1YWdlIjogInB5dGhvbiIsCiAgICAibGlicmFyeV90eXBlIjogIkdBUElDX0FVVE8iLAogICAgInJlcG8iOiAiZ29vZ2xlYXBpcy9weXRob24tZGxwIiwKICAgICJkaXN0cmlidXRpb25fbmFtZSI6ICJnb29nbGUtY2xvdWQtZGxwIiwKICAgICJhcGlfaWQiOiAiZGxwLmdvb2dsZWFwaXMuY29tIiwKICAgICJyZXF1aXJlc19iaWxsaW5nIjogdHJ1ZSwKICAgICJkZWZhdWx0X3ZlcnNpb24iOiAidjIiLAogICAgImNvZGVvd25lcl90ZWFtIjogIiIKfQ==',
+      });
+
+    assert.deepStrictEqual(await owlBotTemplateChanges.checkPR(), false);
+    fileRequest.done();
+  });
+
+  it('should return false in checkPR if incoming PR includes feat', async () => {
+    const owlBotTemplateChanges = new OwlBotTemplateChanges(
+      'gcf-owl-bot[bot]',
+      'feat: a new PR',
+      3,
+      [{filename: 'hello', sha: '2345'}],
+      'testRepoName',
+      'testRepoOwner',
+      1,
+      octokit,
+      'body'
+    );
+
+    const fileRequest = nock('https://api.github.com')
+      .get('/repos/testRepoOwner/testRepoName/contents/.repo-metadata.json')
+      .reply(200, {
+        name: '.repo-metadata.json',
+        content:
+          'ewogICAgIm5hbWUiOiAiZGxwIiwKICAgICJuYW1lX3ByZXR0eSI6ICJDbG91ZCBEYXRhIExvc3MgUHJldmVudGlvbiIsCiAgICAicHJvZHVjdF9kb2N1bWVudGF0aW9uIjogImh0dHBzOi8vY2xvdWQuZ29vZ2xlLmNvbS9kbHAvZG9jcy8iLAogICAgImNsaWVudF9kb2N1bWVudGF0aW9uIjogImh0dHBzOi8vY2xvdWQuZ29vZ2xlLmNvbS9weXRob24vZG9jcy9yZWZlcmVuY2UvZGxwL2xhdGVzdCIsCiAgICAiaXNzdWVfdHJhY2tlciI6ICIiLAogICAgInJlbGVhc2VfbGV2ZWwiOiAiZ2EiLAogICAgImxhbmd1YWdlIjogInB5dGhvbiIsCiAgICAibGlicmFyeV90eXBlIjogIkdBUElDX0FVVE8iLAogICAgInJlcG8iOiAiZ29vZ2xlYXBpcy9weXRob24tZGxwIiwKICAgICJkaXN0cmlidXRpb25fbmFtZSI6ICJnb29nbGUtY2xvdWQtZGxwIiwKICAgICJhcGlfaWQiOiAiZGxwLmdvb2dsZWFwaXMuY29tIiwKICAgICJyZXF1aXJlc19iaWxsaW5nIjogdHJ1ZSwKICAgICJkZWZhdWx0X3ZlcnNpb24iOiAidjIiLAogICAgImNvZGVvd25lcl90ZWFtIjogIiIKfQ==',
+      });
+
+    assert.deepStrictEqual(await owlBotTemplateChanges.checkPR(), false);
+    fileRequest.done();
+  });
+
+  it('should return false in checkPR if incoming PR includes PiperOriginId', async () => {
+    const owlBotTemplateChanges = new OwlBotTemplateChanges(
+      'gcf-owl-bot[bot]',
+      'a fine title',
+      3,
+      [{filename: 'hello', sha: '2345'}],
+      'testRepoName',
+      'testRepoOwner',
+      1,
+      octokit,
+      'PiperOrigin-RevId'
```

Comment:
We could include more text in the body here to test the scenario where there are other things in the body.